core/constants/src/mill/constants/EnvVars.java

```diff
@@ -52,4 +52,8 @@ public class EnvVars {
    * the local disk and not from Maven Central
    */
   public static final String MILL_BUILD_LIBRARIES = "MILL_BUILD_LIBRARIES";
+  /**
+   * Used to store free ports for Mill's tests to use.
+   */
+  public static final String MILL_TEST_FREE_PORT = "MILL_TEST_FREE_PORT";
```

Comment:
Is this a single port or multiple. The comment is a bit vague. If it is multi, that the variable should be `MILL_TEST_FREE_PORTS` and we should document here, which separator is used.